android/src/main/java/com/reactnativereceivesharingintent/ReceiveSharingIntentHelper.java

```diff
@@ -203,30 +200,31 @@ public WritableMap getMediaUris(Intent intent, Context context) {
           queryResult.getColumnIndex(OpenableColumns.DISPLAY_NAME)
         )
       );
+      file.putString(
+        "fileSize",
+        queryResult.getString(queryResult.getColumnIndex(OpenableColumns.SIZE))
+      );
       file.putString("filePath", filePath);
       file.putString("contentUri", contentUri.toString());
       file.putString("text", null);
       file.putString("weblink", null);
       file.putString("subject", subject);
 
       WritableMap exif = getExif(contentUri.toString());
-      exif.putString(
-        "DateTimeModified",
-        queryResult.getString(
-          queryResult.getColumnIndex(MediaStore.Images.Media.DATE_MODIFIED)
-        )
-      );
-      exif.putString(
-        "ImageOrientation",
-        queryResult.getString(
-          queryResult.getColumnIndex(MediaStore.Images.Media.ORIENTATION)
-        )
-      );
-
-      File tempFile = new File(filePath);
-      long fileSize = tempFile.length();
-      file.putInt("fileSize", Math.toIntExact(fileSize));
-
+      try {
+        exif.putString(
+          "DateTimeModified",
+          queryResult.getString(
+            queryResult.getColumnIndex(MediaStore.Images.Media.DATE_MODIFIED)
+          )
+        );
+        exif.putString(
+          "ImageOrientation",
+          queryResult.getString(
+            queryResult.getColumnIndex(MediaStore.Images.Media.ORIENTATION)
+          )
+        );
```

Comment:
I can doublecheck to be sure, but I don't think these two exist in the exif object, since `getExif` is looping over `EXIF_ATTRIBUTES` and they' aren't there

I didn't see a tag that could replace `DATE_MODIFIED` from [available exif tags](https://developer.android.com/reference/androidx/exifinterface/media/ExifInterface) (It's possible I missed something since there's a lot of tags)

For orientation, I recall there were some inconsistencies between different devices and orientation values, but I can update it to use Exif tags instead, I'm hoping upgrade to AndroidX library fixed wrong value issues.

Reply:
Only tags might fit are 

![image](https://github.com/celebrate-app/react-native-receive-sharing-intent/assets/8876645/d6ad9c35-7971-4d93-8497-f05cf568eb24)

Not sure if by modified we really want to have the modified or created date. 

I leave the judgement to you :) 

Thanks for the info anyway

Reply:
I would recommend to take a look into Research I did 3 years ago
Checked attached docs
https://kartenmacherei.atlassian.net/jira/software/projects/CELPHO/issues/CELPHO-660?jql=project%20%3D%20%22CELPHO%22%20AND%20text%20~%20%22exif%22%20ORDER%20BY%20created%20DESC

Basically depends on exif version we either have data or don't

@swizes @ltatarev